data/core/macros/weapon_specials.cfg

```diff
@@ -121,31 +121,33 @@
     [/chance_to_hit]
 #enddef
 
-#define WEAPON_SPECIAL_NAVAL_RAM
+#define WEAPON_SPECIAL_NAVAL
     # Canned definition of the Naval Ram ability to be included in a
     # [specials] clause.
+    [dummy]
+        id=naval
+        name=_"naval"
+        description=_"This attack can only be used when both the attacker and the defender are in water or swamp."
+        special_note=_ "This unit has an attack that can only be used when both the attacker and the defender are in water or swamp."
+    [/dummy]
+    [disable]
+        [filter_self]
+            [filter_location]
+                [not]
+                    terrain="W*^*,S*^*,Km*^*,Cm*^*,Chw^*,Chs^*"
```

Comment:
it is good time to think if https://github.com/wesnoth/wesnoth/pull/9979 can be used